Autofill Management View UI changes

amddg44 opened the pull request:
<!--
Note: This checklist is a reminder of our shared engineering expectations. Feel free to change it, although assigning a GitHub reviewer and the items in bold are required.
-->

Task/Issue URL: https://app.asana.com/0/72649045549333/1202809065048065/f
Tech Design URL:
CC: @THISISDINOSAUR 

**Description**:
Updates tackling all outstanding items except notes. Also includes dark mode support

Note that layout is only focussed on iPhone portrait for now, will tackle landscape + iPad support in another PR 

<!--
Tagging instructions
If this PR isn't ready to be merged for whatever reason it should be marked with the `DO NOT MERGE` label (particularly if it's a draft)
If it's pending Product Review/PFR, please add the `Pending Product Review` label.

If at any point it isn't actively being worked on/ready for review/otherwise moving forward (besides the above PR/PFR exception) strongly consider closing it (or not opening it in the first place). If you decide not to close it, make sure it's labelled to make it clear the PRs state and comment with more information.
-->

**Steps to test this PR**:
1. Run through the various UI states in the management view including new, edit and view ensuring everything is in line with designs
2. Update a password in edit mode 
3. Confirm Open Website button is available + works when website URL is a valid URL 
4. Test deleting a login (and undoing it)
5. Also review in dark mode

<!--
Before submitting a PR, please ensure you have tested the combinations you expect the reviewer to test, then delete configurations you *know* do not need explicit testing.

Using a simulator where a physical device is unavailable is acceptable.
-->

**Copy Testing**:

* [ ] Use of correct apostrophes in new copy, ie `’` rather than `'`

**Orientation Testing**:

* [ ] Portrait
* [ ] Landscape

**Device Testing**:

* [ ] iPhone SE (1st Gen)
* [ ] iPhone 8
* [ ] iPhone X
* [ ] iPad

**OS Testing**:

* [ ] iOS 14
* [ ] iOS 15

**Theme Testing**:

* [ ] Light theme
* [ ] Dark theme

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
**When ready for review, remember to post the PR in MM**

amddg44 committed:
Styling updates in line with designs

amddg44 committed:
Styling updates in line with designs, copy updates, dark mode support, updated assets

amddg44 committed:
Updating a password in edit mode functionality

amddg44 committed:
Updating a password in edit mode functionality

amddg44 committed:
Delete a login while in management view

amddg44 committed:
Update to use AutofillInterfaceEmailTruncator.truncateEmail for username

amddg44 committed:
Menu option to open + navigate to website functionality implemented

amddg44 committed:
Autofill text update

amddg44 committed:
Background styling update

amddg44 committed:
Merge remote-tracking branch 'origin/develop' into feature/anya/autofill-mgmt-ui-changes

THISISDINOSAUR commented on Core/StringExtension.swift:
```diff
@@ -87,3 +87,11 @@ extension String {
+// MARK: - URL
+
+extension String {
+    public var url: URL? {
```
I think this could be confusing, the variable name hides the processing that's going on

THISISDINOSAUR commented on DuckDuckGo/AutofillLoginDetailsView.swift:
```diff
@@ -261,13 +368,17 @@ private struct Copyable: ViewModifier {
     let menuClosedAction: () -> Void
     
-    internal init(isSelected: Bool, menuTitle: String, menuAction: @escaping () -> Void, menuOpenedAction: @escaping () -> Void, menuClosedAction: @escaping () -> Void) {
+    internal init(isSelected: Bool, menuTitle: String, menuAction: @escaping () -> Void, menuSecondaryTitle: String?, menuSecondaryAction: (() -> Void)?, menuOpenedAction: @escaping () -> Void, menuClosedAction: @escaping () -> Void) {
```
Nit: A matter of personal preference since I don't think we have any established pattern on it, but I prefer all the closures to be at the end of the parameter list

THISISDINOSAUR commented on DuckDuckGo/MenuControllerView.swift:
```diff
@@ -59,12 +67,16 @@ struct MenuControllerView<Content: View>: UIViewControllerRepresentable {
         private let onClose: (() -> Void)?
         
-        init(title: String, action: @escaping () -> Void, onOpen: (() -> Void)?, onClose: (() -> Void)?) {
+        init(title: String, action: @escaping () -> Void, secondaryTitle: String?, secondaryAction: (() -> Void)?, onOpen: (() -> Void)?, onClose: (() -> Void)?) {
```
Same as above re argument order

THISISDINOSAUR reviewed (commented):
Just a few very minor nits on the code side.
In terms of testing, a lot of this (even all of it) probably wasn't introduced by this PR, but I did notice it whilst testing:

1. If there's no password, in edit mode, I can't select the password field. I'm guessing this is because it's not editable until you press the reveal button, but when it's empty like this the button never shows.
1. Spacing is off on username/password/website URL above the titles
1. Navigigation bar style is wrong. Looks like this is also a problem in the settings list. Is this becuase of the pending settings redesign?
1. Back button should read Back, not "Autofill Logins"
1. If I press cancel, fields don't revert to old values

amddg44 committed:
Reordering of parameter list

amddg44 committed:
BarButtonItem styles updated

amddg44 committed:
Update row insets

amddg44 committed:
Improvements to edit mode + empty password handling

amddg44 committed:
renamed variable to make functionality more explicit

amddg44 committed:
Reset UI fields state when user cancels while editing

amddg44 committed:
iOS 16 list background styling + navBar styling improvements

amddg44 committed:
iOS 14 list style fix

amddg44 committed:
Merge remote-tracking branch 'origin/develop' into feature/anya/autofill-mgmt-ui-changes

THISISDINOSAUR reviewed (approved):
LGTM!

THISISDINOSAUR committed:
Add notes to autofill logins (#1321)

* Before processing result make sure speech recognition task is alive and not cancelled (#1318)

Task/Issue URL: https://app.asana.com/0/414709148257752/1202599139452865/f

Description:
Due to changes introduced in on-device speech recognition framework the result handler for SFSpeechRecognitionTask appears to be called even after it is cancelled. Those extraneous calls are with empty string result and thus override the query triggering loading of empty SERP.
To avoid this scenario we should check if SpeechRecognizer's internal state is coherent: that the speech recognition task is not cancelled and still assigned to a property before proceeding with processing the result. Otherwise the results are after reported after cancelling the task.

* Add notes field to mangement view

* Do not refresh page on Storage Cache update (#1308)

* Make notes field multiline

* Ensure the favicon size when refreshing from the network (#1319)

Task/Issue URL: https://app.asana.com/0/414235014887631/1202381793018418/f

Description:
Fix for one scenario where the max favicon size is not checked.

* Open Shortcuts app URLs with a confirmation dialog (#1320)

Task/Issue URL: https://app.asana.com/0/414235014887631/1203051209824446/f

Description:
We define platform schemes that prompt user for confirmation as opening those URLs redirects user to another app. iOS provides a shortcuts scheme for launching the Shortcuts app which should be handled in the same way.

* Set version to 7.70.4

* Update embedded files

* Adjust padding

* Remove some unnecssary hstacks

* Remove notes height limit

* Fix notes copied toast text

* Adjust vertical padding when editing notes

Co-authored-by: Michal Smaga <miasma13@gmail.com>
Co-authored-by: bwaresiak <bartek@duckduckgo.com>

amddg44 committed:
Merge remote-tracking branch 'origin/develop' into feature/anya/autofill-mgmt-ui-changes

# Conflicts:
#	Configuration/Version.xcconfig
#	Core/AppPrivacyConfigurationDataProvider.swift
#	Core/ios-config.json
#	DuckDuckGo.xcodeproj/project.pbxproj
#	DuckDuckGo/Favicons.swift
#	DuckDuckGo/Settings.bundle/Root.plist

amddg44 committed:
Styling update to navbar

amddg44 committed:
Removal of iOS 16 modifier for now

amddg44 committed:
Merge remote-tracking branch 'origin/develop' into feature/anya/autofill-mgmt-ui-changes

# Conflicts:
#	DuckDuckGo.xcodeproj/project.pbxproj

amddg44 committed:
Bump BSK to add notes to autofill logins
